### PR TITLE
ZCS-1548 Tabbing doesn't work consistently on all browsers.

### PR DESCRIPTION
**Pull request by jasonyi-zimbra**
ZCS-1548 IE, Edge, and Firefox don't always recognize "onkeypress".
Fix - for IE, Edge, and Firefox, use "onkeydown" instead.

### CODE REVIEW

**Comment by silentsakky** [comment]
Better to put zcs ticket number in commit message and pull request description

File: WebRoot/js/ajax/dwt/xforms/XFormItem.js
Comment (silentsakky) [inline WebRoot/js/ajax/dwt/xforms/XFormItem.js]: I think (AjxEnv.isIE || AjxEnv.isModernIE)  should include mostly all versions of IE

File: WebRoot/js/ajax/dwt/xforms/XFormItem.js
Comment (SatyamKumarOnline) [inline WebRoot/js/ajax/dwt/xforms/XFormItem.js:999]: DwtEvent.ONKEYPRESS can be used here, rather than creating a new variable.

File: WebRoot/js/ajax/dwt/xforms/XFormItem.js
Comment (silentsakky) [reply WebRoot/js/ajax/dwt/xforms/XFormItem.js]: same goes for firefox versions
can we simplify the conditions?

Comment (jasonyi-zimbra) [reply WebRoot/js/ajax/dwt/xforms/XFormItem.js]: I agree that it's bit long but I rather be safe and have a full list.

File: WebRoot/js/ajax/dwt/xforms/XFormItem.js
Comment (jasonyi-zimbra) [reply WebRoot/js/ajax/dwt/xforms/XFormItem.js:999]: Not sure what you mean, but DwtEvent.ONKEYPRESS has different event function associated with it.
"onkeypress"/"onkeydown" attributes  are for AjxEnv specific and have their own event functions.

File: WebRoot/js/ajax/dwt/xforms/XFormItem.js
Comment (silentsakky) [reply WebRoot/js/ajax/dwt/xforms/XFormItem.js]: I would prefer to have cleaner and more understandable code, and most of the conditions are very old and not needed (like IE3, IE4, IE4up etc)

**Review by silentsakky — APPROVED** [review]
Squash commits before merging

**Review by kalpeshhpatel — APPROVED** [review]

**Comment by grishick** [comment]
@silentsakky please merge this pull request. @jasonyi-zimbra does not have the permissions to do the merge and we should treat this PR as any 3rd party contribution. This means that it is up to Zimbra's tech leads to merge PRs.

**Comment by silentsakky** [comment]
Thanks @grishick to let us know, I was waiting for jason to merge the changes :)

Merged in https://github.com/Zimbra/zm-ajax/commit/975e38cc12db79ffe4ff77e927099a3f964f5272

**Pull request closed without merging**
